lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -187,14 +190,12 @@ public Location parentDirectory()
     {
         // todo should this only be allowed for file locations?
         verifyValidFileLocation();
-        checkState(!path.isEmpty(), "root location does not have parent: %s", location);
+        checkState(!path.isEmpty() && !path.equals("/"), "root location does not have parent: %s", location);
 
         int lastIndexOfSlash = path.lastIndexOf('/');
         if (lastIndexOfSlash < 0) {
             String newLocation = location.substring(0, location.length() - path.length() - 1);
-            if (newLocation.isEmpty()) {
-                newLocation = "/";
-            }
+            newLocation += "/";
```

Comment:
Why did the `.isEmpty()` check get removed? It's not immediately clear from the surrounding code or the other changes in this commit.

Reply:
This is so that parent of `scheme://userInfo@host:1234/name` is `scheme://userInfo@host:1234/`
previously it was `scheme://userInfo@host:1234`, but that's forbidden now